### ISSUE TITLE
Speedreader is broken on Figma blog post

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#34959]
## Description 

Speedreader is broken on Figma blog post

## Steps to Reproduce

   1. install `1.63.49`
   2. launch Brave
   3. load `https://www.figma.com/blog/figma-adobe-abandon-proposed-merger/`
   4. click on the `Speedreader` icon

## Actual result:

<img width="1312" alt="Screen Shot 2023-12-18 at 8 33 06 AM" src="https://github.com/brave/brave-browser/assets/387249/9200fc14-0784-428f-b54c-a15e96e57a75">

## Expected result:

Speedreader-formatted article

## Reproduces how often: 

100%
## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @rebron @boocmp @remusao

**Comment by boocmp** [comment]
This is because of the inline style with an absolute position. We need to remove all style attributes from the distilled page.

**Comment by boocmp** [comment]
I've took a closer look and realized that we already have code that removes all visual attributes. But in some cases it doesn't have a chance to execute because of a few strange bugs in our implementation. I've compared  it with the reference implementation and it works fine. After fixing the bugs I've got this 
![image](https://github.com/brave/brave-browser/assets/73575789/ad9215fd-dd41-4a5a-8f19-037a8eae0ee3).
1. I believe that there should not be a list of links at the beginning. I need to figure out why it appears and how to remove it w/o breaking other pages. Our implementation has custom rules for lists, so smth like 'remove a list containing many links' should be ok.
2. After fixing the handler of the 'section' tag, I had a failure in some tests, nothing critical but there may be a better solution that won't cause failures.
